### ISSUE TITLE
Remove the global default scope on models

### ISSUE THREAD

**Issue by cyril** [etalab/admin_api_entreprise#100]
I propose to remove [the global `default_scope`](https://github.com/etalab/admin_api_entreprise/blob/75e95912e1e318c2be36f61864b45e16e3dd3850/app/models/application_record.rb#L4).

There's why:

* https://andycroll.com/ruby/dont-use-default-scope/
* https://www.ombulabs.com/blog/ruby/rails/best-practices/why-using-default-scope-is-a-bad-idea.html
* https://dev.to/lewuathe/why-we-should-avoid-defaultscope-in-rails-3ofo

**Comment by skelz0r** [comment]
Why not, but in this case, the default scope is on `created_at`, which is almost the same as integer ids in a classic rails application, and databases like postgresql tends to sort data by insertion date desc.

In my opinion it's not a priority.

**Comment by brindu** [comment]
Yes! As @skelz0r said it's to mimic the sort by insertion date since we use UUID here.
There are no issues with the default scope as it is simply for ordering data and not for filtering, so no implicit surprises to expect from this.

**Comment by cyril** [comment]
I'm not trying to question the fact of ordering by creation date, following the introduction of UUIDs. It's a way to keep the same behavior as before.

My proposal would therefore not be to correct a problem or a bug, since there is none, but rather to refactor the code in order to explicitly apply this same ordering (optionally when it's useful, as [discussed here](https://github.com/etalab/admin_api_entreprise/pull/92#discussion_r603985428)).

The benefit would be to prevent from possible errors by forgetting this implicit ordering when ordering explicitly on another column. For instance by doing this:

```ruby
User.order(:email)
```

I would expect a list of users ordered by email, without having to think about having to read the settings in the `ApplicationRecord` base model to see if I'm correct. But because of the `default_scope`, my explicit order would be ignored.

**Comment by skelz0r** [comment]
Vu le peu de modèles dans cette application, je pense que c'est inutile de se prendre la tête sur la legacy probable ou non de cette insertion de best practice.

Après un rapide parcours du code, on peut en effet retirer ce `default_scope` et ajouter un order sur les 4 index qui sont:

* users (order `created_at desc`)
* incidents (order `created_at desc`)
* roles (order `name` ?)
* jwt_api_entreprises (order `created_at desc`)

Après comme je disais dans mon 1e commentaire, c'est pas une prio et ne change pas grand chose sur le comportement par défaut d'une app rails nude (vu que PG a tendance à sortir les données par date d'insertion, à noter que ce n'est pas en théorie déterministe comme tu le dit @brindu mais qu'en pratique c'est pratiquement tout le temps le même ordre). Si tu veux scout commit be my guest, j'ai aucun avis.
